test/spec/ol/sphere.test.js

```diff
@@ -1,97 +1,173 @@
-// See http://www.movable-type.co.uk/scripts/latlong.html
-// FIXME add tests for offset
-
-
-import _ol_Sphere_ from '../../../src/ol/Sphere.js';
+import {getArea, getDistance, getLength} from '../../../src/ol/sphere.js';
 import WKT from '../../../src/ol/format/WKT.js';
 import GeometryCollection from '../../../src/ol/geom/GeometryCollection.js';
 import LineString from '../../../src/ol/geom/LineString.js';
 import MultiLineString from '../../../src/ol/geom/MultiLineString.js';
 import MultiPoint from '../../../src/ol/geom/MultiPoint.js';
 import Point from '../../../src/ol/geom/Point.js';
-import _ol_proj_EPSG4326_ from '../../../src/ol/proj/EPSG4326.js';
-
 
-describe('ol.Sphere', function() {
 
-  var sphere = new _ol_Sphere_(6371);
-  var expected = [{
-    c1: [0, 0],
-    c2: [0, 0],
-    haversineDistance: 0
-  }, {
-    c1: [0, 0],
-    c2: [45, 45],
-    haversineDistance: 6671.695598673525
-  }, {
-    c1: [0, 0],
-    c2: [-45, 45],
-    haversineDistance: 6671.695598673525
-  }, {
-    c1: [0, 0],
-    c2: [-45, -45],
-    haversineDistance: 6671.695598673525
-  }, {
-    c1: [0, 0],
-    c2: [45, -45],
-    haversineDistance: 6671.695598673525
-  }, {
-    c1: [45, 45],
-    c2: [45, 45],
-    haversineDistance: 0
-  }, {
-    c1: [45, 45],
-    c2: [-45, 45],
-    haversineDistance: 6671.695598673525
-  }, {
-    c1: [45, 45],
-    c2: [-45, -45],
-    haversineDistance: 13343.391197347048
-  }, {
-    c1: [45, 45],
-    c2: [45, -45],
-    haversineDistance: 10007.543398010286
-  }, {
-    c1: [-45, 45],
-    c2: [-45, 45],
-    haversineDistance: 0
-  }, {
-    c1: [-45, 45],
-    c2: [-45, -45],
-    haversineDistance: 10007.543398010286
-  }, {
-    c1: [-45, 45],
-    c2: [45, -45],
-    haversineDistance: 13343.391197347048
-  }, {
-    c1: [-45, -45],
-    c2: [-45, -45],
-    haversineDistance: 0
-  }, {
-    c1: [-45, -45],
-    c2: [45, -45],
-    haversineDistance: 6671.695598673525
-  }, {
-    c1: [45, -45],
-    c2: [45, -45],
-    haversineDistance: 0
-  }];
-
-  describe('haversineDistance', function() {
+describe('ol/sphere', function() {
+
+  describe('getDistance()', function() {
+
+    var expected = [{
+      c1: [0, 0],
+      c2: [0, 0],
+      distance: 0
+    }, {
+      c1: [0, 0],
+      c2: [45, 45],
+      distance: 6671704.814011975
```

Comment:
For the record: the values differ because of a different radius we used in the tests, right? Looking over the diff this feels odd at first.

Reply:
The more significant difference is that the old `haversineDistance` returned values in kilometers.  The `getDistance` function returns values in meters.  But yes, the numbers are also different because of the different radius.  The old `haversineDistance` used the WGS84 semi-major axis (not really appropriate for a spherical calculation).  The new `getDistance` uses the mean radius of the same ellipsoid (more appropriate for a spherical calculation).